build.wake

```diff
@@ -0,0 +1,416 @@
+
+#
+# Freedom E SDK Target Creation
+#
+
+export tuple FreedomESDKTarget =
+  export SDK:            FreedomESDK
+  export Name:           String
+  export Type:           FreedomESDKTargetType
+  export InstalledFiles: List Path
+
+export data FreedomESDKTargetType =
+  RTLTargetType
+  ArtyTargetType
+  VC707TargetType
+  VCU118TargetType
+  QEMUTargetType
+  HiFiveTargetType
+  SpikeTargetType
+
+def targetTypeToString = match _
+  RTLTargetType  = "rtl"
+  ArtyTargetType = "arty"
+  VC707TargetType = "vc707"
+  VCU118TargetType = "vcu118"
+  QEMUTargetType = "qemu"
+  HiFiveTargetType = "hifive"
+  SpikeTargetType = "spike"
+
+# makeFreedomESDKTargetFromCoreDTS creates a FreedomESDKTarget given:
+#  - sdk: an instance of FreedomESDK, created by makeFreedomESDK
+#  - targetName: the name of the target as a String
+#  - coreDTS: a single Devicetree Source file which describes the hardware
+#             of the target device. This DTS will be used to generate a
+#             Devicetree Overlay
+#  - targetType: one of FreedomESDKTargetType, used to configure the BSP
+#                file generators.
+export def makeFreedomESDKTargetFromCoreDTS sdk targetName coreDTS targetType =
+  def sdkDir = sdk.getFreedomESDKDir
+  def buildEnv = sdk.getFreedomESDKBuildEnv
+  def bspDir = "{sdkDir}/bsp/{targetName}"
+  def runDir = "{sdkDir}/bsp"
+  def updateTargets = "{runDir}/update-targets.sh"
+
+  def cmdline =
+    def venvPath = relative runDir (freedomESDKVenvDir buildEnv)
+    def updateTargetsScript = relative runDir updateTargets
+    def targetDTS = relative runDir coreDTS.getPathName
+    def sdkPath = relative runDir sdkDir
+    "bash", "-c", "%
+      export FREEDOM_E_SDK_VENV_PATH=$(pwd)/%{venvPath}
+      touch ${FREEDOM_E_SDK_VENV_PATH}/.stamp
+      $(pwd)/%{updateTargetsScript} \
+        --target-name %{targetName} \
+        --target-dts=%{targetDTS} \
+        --sdk-path=%{sdkPath} \
+        --target-type %{targetType.targetTypeToString}
+      %", Nil
+
+  def inputs =
+    mkdir bspDir,
+    coreDTS,
+    sdk.getFreedomESDKInstalledFiles
+    ++ (installFreedomESDKVirtualenv buildEnv).getFreedomESDKVirtualenvInstalledFiles
+
+  def fnOutputs _ =
+    def alwaysOutputs =
+      "{bspDir}/core.dts",
+      "{bspDir}/design.dts",
+      "{bspDir}/design.svd",
+      "{bspDir}/metal.default.lds",
+      "{bspDir}/metal.ramrodata.lds",
+      "{bspDir}/metal.scratchpad.lds",
+      "{bspDir}/metal.freertos.lds",
+      "{bspDir}/settings.mk",
+      Nil
+    match targetType
+      ArtyTargetType = "{bspDir}/openocd.cfg", alwaysOutputs
+      VC707TargetType = "{bspDir}/openocd.cfg", alwaysOutputs
+      VCU118TargetType = "{bspDir}/openocd.cfg", alwaysOutputs
+      HiFiveTargetType = "{bspDir}/openocd.cfg", alwaysOutputs
+      _ = alwaysOutputs
+
+  def installedFiles =
+    makePlan cmdline inputs
+    | setPlanResources buildEnv.getFreedomESDKBuildEnvResources
+    | setPlanFnOutputs fnOutputs
+    | setPlanLocalOnly True
```

Comment:
Do you know why this is a LocalOnly job? We've improved the sandboxing story in Wake 0.19, including finally fixing that annoying `.fuse/` directory leaking into the absolute paths. If possible, I'd like to eliminate almost all of the LocalOnly jobs, since they break out of the safety mechanisms of Wake (and they tend to be the cause of the instability or non-reproducibility of our builds today).